### PR TITLE
[SPARK-9661] [MLLIB] minor clean-up of SPARK-9661

### PR DESCRIPTION
**Pull request by mengxr**
Some minor clean-ups after SPARK-9661. See my inline comments. @MechCoder @jkbradley

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAModel.scala [hunk 1]
```diff
@@ -674,10 +674,9 @@ class DistributedLDAModel private[clustering] (
   }
 
   /** Java-friendly version of [[topTopicsPerDocument]] */
-  def javaTopTopicsPerDocument(
-      k: Int): JavaRDD[(java.lang.Long, Array[Int], Array[java.lang.Double])] = {
+  def javaTopTopicsPerDocument(k: Int): JavaRDD[(java.lang.Long, Array[Int], Array[Double])] = {
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAModel.scala]: `weights` should be primitive array.

File: mllib/src/main/scala/org/apache/spark/mllib/stat/Statistics.scala [hunk 2]
```diff
@@ -221,9 +221,7 @@ object Statistics {
   def kolmogorovSmirnovTest(
       data: JavaDoubleRDD,
       distName: String,
-      params: java.lang.Double*): KolmogorovSmirnovTestResult = {
-    val javaParams = params.asInstanceOf[Seq[Double]]
-    KolmogorovSmirnovTest.testOneSample(data.rdd.asInstanceOf[RDD[Double]],
-      distName, javaParams: _*)
+      params: Double*): KolmogorovSmirnovTestResult = {
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/stat/Statistics.scala]: Scala `Double*` should be sufficient here.

File: mllib/src/test/java/org/apache/spark/mllib/clustering/JavaLDASuite.java [hunk 3]
```diff
@@ -44,9 +46,9 @@
   public void setUp() {
     sc = new JavaSparkContext("local", "JavaLDA");
     ArrayList<Tuple2<Long, Vector>> tinyCorpus = new ArrayList<Tuple2<Long, Vector>>();
-    for (int i = 0; i < LDASuite$.MODULE$.tinyCorpus().length; i++) {
```
Comment (mengxr) [inline mllib/src/test/java/org/apache/spark/mllib/clustering/JavaLDASuite.java]: `$.MODULE$` are not necessary

File: mllib/src/test/java/org/apache/spark/mllib/clustering/JavaLDASuite.java [hunk 4]
```diff
@@ -110,8 +112,8 @@ public void distributedLDAModel() {
     assertEquals(roundedLocalTopicSummary.length, k);
 
     // Check: log probabilities
-    assert(model.logLikelihood() < 0.0);
-    assert(model.logPrior() < 0.0);
```
Comment (mengxr) [inline mllib/src/test/java/org/apache/spark/mllib/clustering/JavaLDASuite.java]: Java's `assert` should not used in unit tests, which could be turned off via JVM flags.

File: mllib/src/test/java/org/apache/spark/mllib/clustering/JavaLDASuite.java [hunk 5]
```diff
@@ -126,8 +128,12 @@ public Boolean call(Tuple2<Long, Vector> tuple2) {
     assertEquals(topicDistributions.count(), nonEmptyCorpus.count());
 
     // Check: javaTopTopicsPerDocuments
-    JavaRDD<scala.Tuple3<java.lang.Long, int[], java.lang.Double[]>> topTopics =
-      model.javaTopTopicsPerDocument(3);
```
Comment (mengxr) [inline mllib/src/test/java/org/apache/spark/mllib/clustering/JavaLDASuite.java]: This check is not sufficient due to runtime type erasure. We need to pick out the elements.

**Comment by jkbradley** [comment]
LGTM, thanks for checking!

**Comment by jkbradley** [comment]
I'll merge with master, branch-1.5

**Pull request closed without merging**